scipy/sparse/linalg/_isolve/tests/test_iterative.py

```diff
@@ -57,6 +62,18 @@ def __repr__(self):
         return f"<{self.name}>"
 
 
+class SingleTest:
+    def __init__(self, A, b, solver, casename, convergence=True):
+        self.A = A
+        self.b = b
+        self.solver = solver
+        self.name = casename + '-' + solver.__name__
+        self.convergence = convergence
+
+    def __repr__(self):
+        return f"<{self.name}>"
```

Comment:
Since we're explicitly setting `id=`, I think the repr isn't necessary anymore actually